### DISCUSSION TITLE
Disconect and not useable on ESP32-S2

### DISCUSSION THREAD

**Discussion by dzungpv** [bertmelis/espMqttClient#138] [General]
I run a project on ESP32-S2, it has single core 2MB PSRAM, frequently disconnect reason 7, It more powerful than ESP8266 but I dont know why. The project running Homekit stack, Async Web-sever and espMqttClient. I check the ram it is free 92% all the time.

**Comment by bertmelis** [comment]
Which S2 board?
Do yo have access to the broker logs?

All due respect for your problem, but the solution is to be found outside this library. There is no platform-specific code inside the library. This doesn't mean I'm not on the lookout for the root cause.

**Reply by dzungpv** [reply]
S2 Mini wemos. I revert back to old mqtt async lib and it work fine. The broker is mine, no tls support. It just disconnected

**Reply by bertmelis** [reply]
So you're using async? Which version in your code do you use? `espMqttClient` or `espMqttClientAsync`?
Other people also reported to have issues with the async version. But on ESP32, my advice is to use the sync version.

Since it's your own broker, could you check the broker's log to see why it is disconnecting?

**Reply by dzungpv** [reply]
It is https://github.com/marvinroger/async-mqtt-client, the old lib I use, with ESP32 your lib stable, but something wrong with ESP32 S2.

**Reply by dzungpv** [reply]
I read your code, it may relate to something like: https://github.com/bertmelis/espMqttClient/blob/885e631f1669aaaeb88098811aace215a43d01b8/src/MqttClient.cpp#L65C17-L65C17 S2 only has 1 core, it is core 0, your default code core = 1 https://github.com/bertmelis/espMqttClient/blob/885e631f1669aaaeb88098811aace215a43d01b8/src/espMqttClient.h#L51 I will try to fix it, better not pinto core, let Free-RTOS decide that

**Reply by bertmelis** [reply]
I forgot about that piece. It indeed pins the task to core 1 by default. I don't know what happens if there's only one core.

It is better to set it to `tskNO_AFFINITY`.

It would surprise me if this renders the lib unstable but it is quickly tested.

**Reply by dzungpv** [reply]
I fixed and tested with this PR, it is fine now.
https://github.com/bertmelis/espMqttClient/pull/139

**Reply by bertmelis** [reply]
Before I merge, let me test me a dual core ESP32 too.
People that specify a specific core won't be affected but I seldomly do that myself.

**Reply by bertmelis** [reply]
With Christmas in a few days, activitiy will be low. So this probably has to wait a few days.

**Reply by dzungpv** [reply]
Yes, but you can see the code, it detect number of core of the ESP32 chip with https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-reference/system/misc_system_api.html#_CPPv415esp_chip_info_t and same with https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/misc_system_api.html#_CPPv415esp_chip_info_t and c3 etc. If chip more than 2 core, the same behavior. I have ESP32 dual core running too.

**Reply by dzungpv** [reply]
But TLS MQTT is the same, it is not fix with my PR, it is somethings else, only normal mqtt work for ESP32S2 chip. May be my app too heavy for the 1 core chip.
